Added Kubernetes-the-hard-way playbook

ardikabs opened the pull request:
Related articles https://github.com/kelseyhightower/kubernetes-the-hard-way

List added roles
- calico-node (for CNI provision)
- docker (for CRI provision)
- kubernetes/master (for master provision)
- kubernetes/worker (for worker provision)

Also we structurize ansible-playgrounds project, like:
- inventory (inventory file, separated by datacenter on each directory tree)
- plays (playbook file, separated by datacenter on each directory tree)
- roles

ardikabs committed:
Added Kubernetes-the-hard-way playbook
Related articles https://github.com/kelseyhightower/kubernetes-the-hard-way

List added roles
- calico-node (for CNI provision)
- docker (for CRI provision)
- kubernetes/master (for master provision)
- kubernetes/worker (for worker provision)

Also we structurize ansible-playgrounds project, like:
- inventory (inventory file, separated by datacenter on each directory tree)
- plays (playbook file, separated by datacenter on each directory tree)
- roles

dimasmamot commented on inventory/dc1/group_vars/all:
```diff
@@ -0,0 +1,2 @@
+ansible_user: CHANGE_WITH_YOUR_ANSIBLE_USER_HERE
```
Majority jenis instance di satu datacenter yang sama, punya identifier yang sama, jadi haruse ngga perlu di `CHANGE_WITH_YOUR_ANSIBLE_USER_HERE`, langsung di spesifikan ae lek jareku, soale bakal ribet pas nanti provision ansible executor baru, harus edit edit

dimasmamot commented on inventory/dc1/group_vars/all:
```diff
@@ -0,0 +1,2 @@
+ansible_user: CHANGE_WITH_YOUR_ANSIBLE_USER_HERE
+ansible_ssh_private_key_file: CHANGE_WITH_YOUR_SSH_PRIVATE_KEY_FILE
```
Iki rencana ditaro mana ? mending saranku, bikin satu directory, di level root ansible, directory `key` isinya kosong, isinya key, tapi ngga di push ke repo, ekslusif di instance ditambahnya, dan langsung di vars `all` ini di spesifikan, key yang mana yang bakal dipake, tujuan e tetep, biar seamless ga perlu edit lek clone repo ansible iki ke ansible manager baru

ardikabs replied:
jadi artinya, tiap menjalankan playbook mending di specify `ansible_user` aja ? instead ditaruh di `group_vars` ya ?

dimasmamot commented on plays/dc1/k8s-bootstrap.yml:
```diff
@@ -0,0 +1,38 @@
+---
```
Lek iki, prasaku kok mending stage `Ensure python is present` iku langsung jadikan role `common` jadi di playbook bersih, ngga ada task tambahan

```
---
- name: Setup kubernetes bootstrap
  hosts: all
  become: yes
  vars:
    wawa: val
    qaaqa: val
    sad: [ ls1, ls2, ls3 ]

  roles:
    - common
    - role2
    - role3
```

tujuane cek modular ae, bersih playbook e, dan konfigurasi untuk ensure ensure, pre dependency iku terkontrol nde satu file, role

ardikabs replied:
okee, will be added as role

dimasmamot commented on plays/dc1/swarm-bootstrap.yml:
```diff
@@ -0,0 +1,24 @@
+---
```
Nah koyok ngene, awamu kan nulis maneh `Ensure python is present`

mending langsung dadino role

terus panggil sesuai urutan

dimasmamot commented on roles/calico-node/tasks/generate_certs.yml:
```diff
@@ -0,0 +1,25 @@
+  - "{{tls_ca}}"
+  - "{{tls_ca_key}}"
+  - "{{tls_ca_config}}"
+
```
`tags` jangan lupakan tags, berguna lek misale awamu pengen ngajalano beberapa `stage` nde role iki tanpa harus eksekusi semua `stage` jadi pas ansible-playbook tinggal nggae `--tags`

ardikabs replied:
nice idea

dimasmamot reviewed (changes requested):
Ngga akeh sing isok di review, aku gapaham roles e soale wkwkkw kakean, cuma 5 iku prasaku isok lumayan oke

ardikabs committed:
Added README.md

ardikabs committed:
Added tags

ardikabs committed:
Added common role

ardikabs committed:
Added nginx role

ardikabs committed:
Minor changes on kubernetes/master role

ardikabs committed:
minor changes

ardikabs committed:
Added support for kubernetes version v1.10.13
Separated kubernetes services based on version (templates/services-{{ kube version }})

ardikabs committed:
ReStructure calico

ardikabs committed:
Fixing variable for master and worker
